packages/api/src/middleware/tracking.ts

```diff
@@ -60,13 +61,40 @@ class Tracker {
     }
   }
 
-  async flushAll() {
-    const all = this.pendingUpdates;
-    this.pendingUpdates = new Map();
+  pendingWebhookStatusUpdates: Map<string, DBWebhook["status"]> = new Map();
+
+  recordWebhookStatus(id: string, status: DBWebhook["status"]) {
+    const key = `${id}`;
+    const alreadyScheduled = this.pendingWebhookStatusUpdates.has(key);
+    this.pendingWebhookStatusUpdates.set(key, status);
+    if (alreadyScheduled) return;
+
+    setTimeout(() => this.flushWebhookStatus(key), flushDelay);
+  }
 
-    for (const key of all.keys()) {
+  private async flushWebhookStatus(id: string) {
+    const status = this.pendingWebhookStatusUpdates.get(id) ?? {};
```

Comment:
We should probably not have this `?? {}` default, otherwise line 78 will never be true and we might update the webhook with an empty status.
```suggestion
    const status = this.pendingWebhookStatusUpdates.get(id);
```